server/sonar-server/src/test/java/org/sonar/server/organization/DefaultOrganizationProviderRule.java

```diff
@@ -34,9 +50,22 @@ public static DefaultOrganizationProviderRule create(DbTester dbTester) {
     return new DefaultOrganizationProviderRule(dbTester);
   }
 
-
   @Override
   public DefaultOrganization get() {
-    return new DefaultOrganizationProviderImpl(dbTester.getDbClient()).get();
+    DbSession dbSession = dbTester.getSession();
+    Optional<String> uuid = dbTester.getDbClient().internalPropertiesDao().selectByKey(dbSession, InternalProperties.DEFAULT_ORGANIZATION);
+    checkState(uuid.isPresent() && !uuid.get().isEmpty(), "No Default organization uuid configured");
+    Optional<OrganizationDto> dto = dbTester.getDbClient().organizationDao().selectByUuid(dbSession, uuid.get());
+    checkState(dto.isPresent(), "Default organization with uuid '%s' does not exist", uuid.get());
+    return toDefaultOrganization(dto.get());
+  }
+
+  public OrganizationDto getDto() {
```

Comment:
the contract of `DefaultOrganizationProviderRule` is highly confusing to me and I don't really understand what it is meant to achieve : `get()` is stateless while `getDto()` is statefull, which basically means they can be out of sync with each other (a change in DB will be visible by calling get() but not by getDto()`.

As `DefaultOrganizationProvider` does not enforce any caching, I think `DefaultOrganizationProviderRule` should either be fully stateless or full stateful (if both behaviors are actually used in practice). If so, factory methods should be created for each implementation.